onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -317,30 +404,67 @@ TensorrtExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
-  const char* batch_env = getenv("ORT_TENSORRT_MAX_PARSER_ITERATIONS");
-  const int max_iterations = batch_env ? atoi(batch_env) : max_parser_iterations_;
   bool early_termination = false;
-  supported_nodes_vector = GetSupportedList(parser_nodes_vector, 0, max_iterations, graph, &early_termination);
+  supported_nodes_vector = GetSupportedList(parser_nodes_vector, 0, max_partition_iterations_, graph, &early_termination);
   if (early_termination) {
     supported_nodes_vector.clear();
   }
 
+  // Remove nodes with empty shape (for example [1, 0]) because TensorRT 6 doens't support empty shape
+  SubGraphCollection_t post_processed_supported_nodes_vector;
+  for (auto& group : supported_nodes_vector) {
+    // Right now only NonZero and NonMaxSuppression related empty shape nodes are removed.
+    // The typical cases are Faster-rcnn and Mask-rcnn
+    // TODO: Remove the code if TensorRT fixed the issue in future release, or find a better generic way here to work around
+    post_processed_supported_nodes_vector.push_back({});
+    for (const auto& index : group.first) {
+      const auto& node = graph.GetNode(node_index[index]);
+      bool exclude_node = false;
+      for (auto input : node->InputDefs()) {
+        auto input_shape = input->Shape();
+        if (input_shape) {
+          for (auto dim : input_shape->dim()) {
+            std::string dim_name = dim.dim_param();
+            std::string exclude_dim_name1 = "NonZero";
+            std::string exclude_dim_name2 = "NonMaxSuppression";
+            if (!dim_name.empty()) {
+              if ((dim_name.find(exclude_dim_name1) != std::string::npos) || (dim_name.find(exclude_dim_name2) != std::string::npos)) {
+                exclude_node = true;
+                break;
+              }
+            }
+          }
+        }
+
+        if (exclude_node) {
+          break;
+        }
+      }
+      if (!exclude_node) {
+        post_processed_supported_nodes_vector.back().first.push_back(index);
+      }
+    }
+
+    // Remove subgraph if it is empty or its size is smaller than the predefined minimal subgraph size
+    const int subgraph_size = post_processed_supported_nodes_vector.back().first.size();
+    if (subgraph_size == 0 || subgraph_size < min_subgraph_size_) {
+      post_processed_supported_nodes_vector.pop_back();
+    } else {
+      post_processed_supported_nodes_vector.back().second = group.second;
+    }
+  }
+
   // Construct subgraph capability from node list
   std::vector<std::unique_ptr<ComputeCapability>> result;
   int counter = 0;
-  for (const auto& group : supported_nodes_vector) {
+  for (const auto& group : post_processed_supported_nodes_vector) {
     if (!group.first.empty()) {
       std::unique_ptr<IndexedSubGraph> sub_graph = GetSubGraph(group, counter, graph);
-      result.push_back(onnxruntime::make_unique<ComputeCapability>(std::move(sub_graph)));
+      result.push_back(std::make_unique<ComputeCapability>(std::move(sub_graph)));
```

Comment:
I think we should use onnxruntime::make_unique instead of std::make_unique

Reply:
good catch. Will change